python/src/scipp/plot/plot.py

```diff
@@ -25,6 +28,19 @@ def plot(scipp_obj,
     # Delayed imports
     from .tools import get_line_param
     from .dispatch import dispatch
+    import matplotlib.pyplot as plt
+    import matplotlib as mpl
+
+    # Determine if we are using a static or interactive backend
+    interactive = mpl.get_backend().lower().endswith('nbagg')
+
+    # Switch interactive plotting off for better control over when figures are
+    # displayed.
+    # TODO: we need to think about what users expect, if they than have custom
+    # figures further down in the notebook. We would want to switch
+    # interactivity back on, but then we would have to turn it off again every
+    # time a SciPlot slider is moved or a button is pressed.
+    plt.ioff()
```

Comment:
To what extent does scipp support returning "standard" mpl figures for users who are already familiar with matplotlb and want to script a plot to look like their specifications?

Reply:
yes, that is a good point :+1: 
I need to switch it back on afterwards, and i tried doing that but i was getting some strange results. Some scipp figures somehow ended up being cached and suddently appeared when i made a normal totally independent matplotlib figure.

i'll have another go

Reply:
So a solution to this one would be to check whether we are in interactive mode when `plot` is called, and then run `plt.ion()` at the end of the `plot` function to turn interactivity back on, in case interactive mode was on prior to calling `plot`. You would do this with
```Py
import matplotlib as mpl
mpl.is_interactive()
```

(here, "interactivity" means that when you create a matplotlib figure, it automatically shows up in the notebook without having to explicitly show the figure. We are not talking about interactivity with zoom or sliders)

Hence, when i'm in a notebook, and the first thing i do is import `scipp`, and then plot something with it, it would find that interactivity is off (because matplotlib interactivity only get enabled when `matplotlib.pyplot` is imported). So after a call to plot, it would not turn interactivity back on because it was not on in the first place.

However, if in a cell below i would like to make my own matplotlib figure,
```Py
import matplotlib.pyplot as plt
fig, ax = plt.subplots()
```
doesn't automatically show the figure, and
```Py
mpl.is_interactive()
```
returns `False`
I think this is because `matplotlib.pyplot` has already been imported by `scipp` and so this subsequent explicit import does not turn interactivity back on.

Reply:
Another option would be to always enable interactive plotting with `plt.ion()` after the plot, and just assume that it will be rare that a user will have disabled interactive mode with `plt.ioff()` prior to plotting with `scipp`.